contracts/gateway/EIP20Gateway.sol

```diff
@@ -488,7 +496,42 @@ contract EIP20Gateway is GatewayBase {
             bytes32(0),
             true
         );
+    }
+
+    /**
+     * @notice Return the revert penalty to the staker. Only valid for
+     *         a message transition from DeclaredRevocation -> Progressed.
+     *
+     * @dev Should only be called from progressStakeWithProof. This function
+     *      exists to avoid a stack too deep error.
+     *
+     * @param _messageHash Message hash.
+     * @param _staker Staker address.
+     * @param _oldMessageStatus Message status before progressing.
+     * @param _newMessageStatus Message status after progressing.
+     */
+    function tryReturnPenaltyToStaker(
+        bytes32 _messageHash,
+        address _staker,
+        MessageBus.MessageStatus _oldMessageStatus,
+        MessageBus.MessageStatus _newMessageStatus
```

Comment:
This is inbox message status, right?

Reply:
Logic will be more readable with this.  Return penalty to the staker if outbox message status is `DeclaredRevocation` and inbox message status is `Progressed`.